### ISSUE TITLE
Add strict option to scheduler

### ISSUE THREAD

**Issue by carmocca** [PyTorchLightning/pytorch-lightning#3526]
## 🚀 Feature

When a `ReduceLROnPlateau` scheduler is used, training stops with a raised `MisconfigurationException` if the `monitor` is not found. We should add a `strict` option (same as in the `EarlyStopping` callback) so it is configurable.

EarlyStopping:
https://github.com/PyTorchLightning/pytorch-lightning/blob/c64520e658806a87f282c74299b2ea4b7ea493ea/pytorch_lightning/callbacks/early_stopping.py#L61-L62
https://github.com/PyTorchLightning/pytorch-lightning/blob/c64520e658806a87f282c74299b2ea4b7ea493ea/pytorch_lightning/callbacks/early_stopping.py#L107-L116

OptimizerConnector:
https://github.com/PyTorchLightning/pytorch-lightning/blob/c64520e658806a87f282c74299b2ea4b7ea493ea/pytorch_lightning/trainer/connectors/optimizer_connector.py#L53-L59

### Motivation

If my first epoch fails to produce a valid loss, the scheduler stops training.
### Pitch

```python
    def configure_optimizers(self):
        optimizer = ...
        scheduler = {
            "scheduler": torch.optim.lr_scheduler.ReduceLROnPlateau(
                optimizer,
                mode="min",
            ),
            "monitor": "checkpoint_on",
            "interval": "epoch",
            "frequency": 1,
            "strict": False  # Allow this
        }
        return [optimizer], [scheduler]
```

**Comment by Borda** [comment]
> If my first epoch fails to produce a valid loss, the scheduler stops training.

can you elaborate a bit more on this case, not sure what is the case that you would endup without loss...

**Comment by SkafteNicki** [comment]
@carmocca is there a specific reason why your model fails to produce a validation loss for the first epoch?
I like this addition as we have already encountered before that some users setup their training such that sometimes the validation loss is computed after they need it for their scheduler (https://github.com/PyTorchLightning/pytorch-lightning/issues/1598, https://github.com/PyTorchLightning/pytorch-lightning/issues/1156)

**Comment by carmocca** [comment]
Same reason as https://github.com/PyTorchLightning/pytorch-lightning/pull/3527#issuecomment-694827383
